decidim-core/app/packs/src/decidim/input_character_counter.js

```diff
@@ -264,7 +263,7 @@ export default class InputCharacterCounter {
       if (remaining === 1) {
         message = MESSAGES.charactersLeft.one;
       }
-      this.$input[0].dispatchEvent(
+      this.$userInput[0].dispatchEvent(
```

Comment:
I have noticed that we have `this.$input` and `this.$userInput` (maybe the same we have defined in the line 92). what is the difference between those 2 calls?

Reply:
The first one (`this.$input`) refers to the plain input field, which in normal situations would be e.g. input or textarea.

The other one (`this.$userInput`) refers to the contenteditable area, in this case the ProseMirror controlled div. The value of the contenteditable area is not sent to the server but it is added to the hidden input field (i.e. `this.$input`) when the contenteditable area is updated.

In case of a rich text editor, `this.$input` variable refers to the hidden input field containing the value to be sent to the server and `this.$userInput` refers to the contenteditable area. We want to dispatch the event directly to the contenteditable area so that the editor itself can be completely unaware of the hidden input related to it.